srcopsmetrics/bot_knowledge.py

```diff
@@ -18,19 +18,41 @@
 """Create, Visualize, Use bot knowledge from different Software Development Platforms."""
 
 import logging
+import os
+from importlib import import_module
 from pathlib import Path
+from pkgutil import iter_modules
+from typing import List, Optional, Tuple
 
-from typing import List, Tuple, Optional
-
-from srcopsmetrics.enums import EntityTypeEnum
+from srcopsmetrics.entities import Entity
+from srcopsmetrics.exceptions import NotKnownEntities
 from srcopsmetrics.github_knowledge import GitHubKnowledge
 from srcopsmetrics.utils import check_directory
-from srcopsmetrics.exceptions import NotKnownEntities
+
+import inspect
 
 _LOGGER = logging.getLogger(__name__)
 
 github_knowledge = GitHubKnowledge()
 
+NOT_FOR_INSPECTION = {"interface", "template", "tools"}
+
+
+def get_all_entities():
```

Comment:
why not just having a variables called ENTITIES to be imported in `srcopsmetrics/entities/__init__.py `?

Reply:
because this was the intention of having an automatic mechanism that does this without each time adding manually every new entity that is implemented
by this approach the contributor does not have to look anywhere but the entity implementation itself to look for any causes of issues or etc. inside the `mi` application

all of functionality outside the `entities` folder refers to just an automation purposes of collecting knowledge, therefore any specific entities should not be used there in purpose of generality i think, therefore we even don't have to specify new entities in `entities/__init__.py` folder

or am I getting something wrong? Maybe the only issue with this approach is its overhead that everytime an inspection is run, the classes have to be inspected.

Reply:
I mean it's double I think, I just find the other more expressive because I want to see immediately from enums what entities exist, NOT_FOR_INSPECTION need to be updated in any case for something new not as entity

Reply:
> NOT_FOR_INSPECTION need to be updated in any case for something new not as entity

yes, that's true, but the probability of it being updated is really low considering that the `entities` module should only contain entities and nothing more (except some utilities represented in `tools` submodule). On the other hand the case of updating available entities is higher, coz that's the main purpose of mi application. Therefore the contributor does not have to do anything besides pull requesting only the entity file he/she implemented. I think that this is more a DRY-type solution than having each entity specified in __init__ (which in long-term can result of a very long list), but I can be definitely wrong...

I've moved `NOT_FOR_INSPECTION` to the `entities` so that it is more clear where it should belong

but yeah,  i mean if it's really unreasonable to have this functionality here I can definitely pull it back.

Reply:
let's use your solution :)

Reply:
:sake: tonight drink's on me then!